odbc/src/api/data.rs

```diff
@@ -1122,7 +1122,9 @@ pub mod i32_len {
             output_ptr as *mut WideChar,
             buffer_len / size_of::<WideChar>(),
         );
-        *text_length_ptr = (size_of::<WideChar>() * len) as Integer;
+        if !text_length_ptr.is_null() {
```

Comment:
oh, good protection! 👍
Do you now why there is still a bunch of other places in the file where we don't do the check?
Maybe using set_str_length everywhere would actually be the best course of action, since it wraps the affection is a check.